src/components/views/dialogs/DevtoolsDialog.js

```diff
@@ -605,12 +614,97 @@ class ServersInRoomList extends React.PureComponent {
     }
 }
 
+const PHASE_MAP = {
+    [PHASE_UNSENT]: "unsent",
+    [PHASE_REQUESTED]: "requested",
+    [PHASE_READY]: "ready",
+    [PHASE_DONE]: "done",
+    [PHASE_STARTED]: "started",
+    [PHASE_CANCELLED]: "cancelled",
+};
+
+function VerificationRequest({txnId, request}) {
+    const [, updateState] = useState();
+    const [timeout, setTimeout] = useState(request.timeout);
+
+    /* Re-render if something changes state */
+    useEffect(() => {
+        request.on("change", updateState);
+        return () => request.off("change", updateState);
+    }, [request]);
+
+    /* Keep re-rendering if there's a timeout */
+    useEffect(() => {
+        if (request.timeout == 0) return;
+
+        const id = setInterval(() => {
+           setTimeout(request.timeout);
```

Comment:
what is this meant to be doing?

Reply:
keeps the timeout up to date

Reply:
to be more clear, `request.timeout` is defined as such:

```
    /** The current remaining amount of ms before the request should be automatically cancelled */
    get timeout() {
        const requestEvent = this._getEventByEither(REQUEST_TYPE);
        if (requestEvent) {
            const elapsed = Date.now() - this.channel.getTimestamp(requestEvent);
            return Math.max(0, VERIFICATION_REQUEST_TIMEOUT - elapsed);
        }
        return 0;
    }
```

ie it's actually running code

Reply:
oh right its just using the default of 0ms, `setImmediate` might be clearer here

Reply:
ah, `setTimeout` here is a state function, not `window.setTimeout`. Maybe I should clarify that

Reply:
my bad, completely missed that